libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -636,15 +629,6 @@ internal object MapboxNavigationTelemetry : MapboxNavigationTelemetryInterface {
         initializer = preInitializePredicate
     }
 
-    private fun validateAccessToken(accessToken: String?) {
-        if (accessToken.isNullOrEmpty() ||
-            (!accessToken.toLowerCase(Locale.US).startsWith("pk.") &&
-                !accessToken.toLowerCase(Locale.US).startsWith("sk."))
-        ) {
-            throw NavigationException("A valid access token must be passed in when first initializing MapboxNavigation")
-        }
-    }
```

Comment:
> For telemetry, the access token is needed by the `MapboxTelemetryReporter`. It gets its access token through MapboxNavigation
https://github.com/mapbox/mapbox-navigation-android/blob/5d7113ed6fa65cde72c2d57f2ca558e6c8dc3a8b/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt#L197


Should we add the check in there or in `MapboxMetricsReporter`? Should we even check the access token? Requests will fail in the backend so what's the problem? cc @nkukday @harvsu

Reply:
Could consider adding a place for access token validation if it is useful. Related to the work we need to do here https://github.com/mapbox/mapbox-navigation-android/issues/2631